data_hub/views/api_views.py

```diff
@@ -40,3 +43,34 @@ def get(self, request):
             return JsonResponse(json.loads(df), safe=False)
 
         return JsonResponse(serializer.errors, status=400)
+
+
+class MacroeconomicAPIView(APIView):
+    def get(self, request):
+
+        serializer = MacroeconomicAPIParamSerializer(data=request.query_params)
+        data_service = DataService()
+        if serializer.is_valid():
+            name = serializer.validated_data.get("name")
+            frequency = serializer.validated_data.get("frequency")
+            from_date = serializer.validated_data.get("from_date")
+            to_date = serializer.validated_data.get("to_date")
+            year = serializer.validated_data.get("year")
+            month = serializer.validated_data.get("month")
+            quarter = serializer.validated_data.get("quarter")
+
+            df = data_service.get_macroeconomic_data(
+                name, to_date, from_date, year, frequency, month, quarter
+            ).to_json(orient="records")
+
+            return JsonResponse(json.loads(df), safe=False)
+
+        return JsonResponse(serializer.errors, status=400)
+
+
+class MacroeconomicDataSourcesAPIView(APIView):
+    def get(self, request):
+
+        data_service = DataService()
```

Comment:
```suggestion
    def get(self, request):
        data_service = DataService()
```
nit: is better to not start code inside functions with a blank line :)